### PR TITLE
OADP-2048:Add volsync cron expression support

### PR DESCRIPTION
**Pull request by shubham-pampattiwar**
Related VSM controller PR: https://github.com/migtools/volume-snapshot-mover/pull/247

### CODE REVIEW

File: api/v1alpha1/oadp_types.go [hunk 1]
```diff
@@ -218,6 +218,12 @@ type DataMover struct {
 	// defines the parameters that can be specified for retention of datamover snapshots
 	// +optional
 	SnapshotRetainPolicy *RetainPolicy `json:"snapshotRetainPolicy,omitempty"`
+	// schedule is a cronspec (https://en.wikipedia.org/wiki/Cron#Overview) that
+	// can be used to schedule replication to occur at regular, time-based
+	// intervals.
+	//+kubebuilder:validation:Pattern=`^(\d+|\*)(/\d+)?(\s+(\d+|\*)(/\d+)?){4}$`
+	//+optional
+	Schedule string `json:"schedule,omitempty"`
```
Comment (kaovilai) [inline api/v1alpha1/oadp_types.go:226]: How would one setting this setting affect backup or restore with data mover?

I would add description to provide a one usecase example so they can see via `oc explain` and other mechanisms (openshift web console)

Comment (kaovilai) [reply api/v1alpha1/oadp_types.go:226]: This is part of the [requirement in the jira](https://issues.redhat.com//browse/OADP-2048) as well

> Instructions for setup can be found here: `oc explain dpa.Spec.Features.DataMover.Schedule`

**Review by kaovilai — CHANGES_REQUESTED** [review]
please address jira requirement per [my comments](https://github.com/openshift/oadp-operator/pull/1080/files#r1253174022)

File: api/v1alpha1/oadp_types.go (hunk 1 above)
Comment (kaovilai) [reply api/v1alpha1/oadp_types.go:226]: at minimum could link to https://volsync.readthedocs.io/en/stable/usage/triggers.html is my 2cents.

Comment (shubham-pampattiwar) [reply api/v1alpha1/oadp_types.go:226]: Triggers are used for starting the Volsync synchronization process, by default we had manual triggers, this PR helps add a schedule trigger.

Comment (shubham-pampattiwar) [reply api/v1alpha1/oadp_types.go:228]: Done, updated the PR.

**Review by kaovilai — APPROVED** [review]
/lgtm :D
Thank you!

**Review by sseago — APPROVED** [review]

**Comment by shubham-pampattiwar** [comment]
Cherry-pick PR: https://github.com/openshift/oadp-operator/pull/1084

**Pull request merged**